bot/cogs/help_channels.py

```diff
@@ -217,8 +217,8 @@ def create_name_queue(self) -> deque:
                 with suppress(discord.errors.HTTPException, discord.errors.NotFound):
                     await self.reset_claimant_send_permission(ctx.channel)
 
-                self.cancel_task(ctx.channel.id)
                 await self.move_to_dormant(ctx.channel)
+                self.cancel_task(ctx.channel.id)
```

Comment:
On second thought this could be a race condition but I'm not too worried about it.

Reply:
Just to document, could you describe the race condition you were thinking of? Could be relevant if we do notice something going wrong in the future. (Although I'm not worried about that as you're not too worried about that as well.)

`self.cancel_task` should only be called once `move_to_dormant` returns, which I think is after it has done its work (in the current implementation), I think. There are currently no `create_task` or other non-awaited scheduling methods involved.

Reply:
Maybe race condition is the wrong term. If the command is invoked when the channel is about to automatically go dormant, then it may attempt to move an already dormant channel. I am not sure what the consequences of that are. This problem would still exist anyway even if the order was flipped back around. A robust solution would probably require some sort of lock, but that would mean one lock per channel which gets relatively complex.

Reply:
Thanks, I see what you mean. I guess we'll found out, but this is bound to happen occasionally.